cmd/deploy/deploy.go

```diff
@@ -0,0 +1,173 @@
+package deploy
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/google/uuid"
+	contextCMD "github.com/okteto/okteto/cmd/context"
+	"github.com/okteto/okteto/cmd/utils"
+	"github.com/okteto/okteto/pkg/log"
+	"github.com/okteto/okteto/pkg/model"
+	"github.com/spf13/cobra"
+	"k8s.io/client-go/rest"
+)
+
+var tempKubeConfig = "/var/tmp/okteto.kubeconfig"
+
+type kubeConfigHandler interface {
+	Read() (*rest.Config, error)
+	Modify(ctx context.Context, port int, sessionToken string) error
+}
+
+type proxyInterface interface {
+	Start(ctx context.Context, name string, clusterConfig *rest.Config) error
+	Shutdown(ctx context.Context) error
+	GetPort() int
+	GetToken() string
+}
+
+type manifestExecutor interface {
+	Execute(command string, env []string) error
+}
+
+type deployCommand struct {
+	getSecrets  func(ctx context.Context) ([]string, error)
+	getManifest func(cwd, name, filename string) (*Manifest, error)
+
+	proxy      proxyInterface
+	kubeconfig kubeConfigHandler
+	executor   manifestExecutor
+}
+
+//Deploy deploys the okteto manifest
+func Deploy(ctx context.Context) *cobra.Command {
+	var filename string
+	var variables []string
+	var name string
+
+	cmd := &cobra.Command{
+		Use:   "deploy",
+		Short: "Executes the list of commands specified in the Okteto manifest",
+		Args:  utils.NoArgsAccepted("https://okteto.com/docs/reference/cli/#version"),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := contextCMD.Init(ctx); err != nil {
+				return err
+			}
+
+			cwd, err := os.Getwd()
+			if err != nil {
+				return fmt.Errorf("failed to get the current working directory: %w", err)
+			}
+
+			if name == "" {
+				name = getName(cwd)
+			}
+
+			// Look for a free local port to start the proxy
+			port, err := model.GetAvailablePort("localhost")
+			if err != nil {
+				log.Errorf("could not find a free port to start proxy server: %s", err)
+				return err
+			}
+			log.Debugf("found available port %d", port)
+
+			// TODO for now, using self-signed certificates
+			cert, err := tls.X509KeyPair(cert, key)
+			if err != nil {
+				log.Errorf("could not read certificate: %s", err)
+				return err
+			}
+
+			// Generate a token for the requests done to the proxy
+			sessionToken := uuid.NewString()
+
+			c := &deployCommand{
+				getSecrets:  getOktetoSecretsAsEnvironmenVariables,
+				getManifest: getManifest,
+
+				kubeconfig: newKubeConfig(),
+				executor:   newExecutor(),
+				proxy: newProxy(proxyConfig{
+					port:         port,
+					certificates: []tls.Certificate{cert},
+					token:        sessionToken,
+				}),
+			}
+			return c.runDeploy(ctx, cwd, name, filename, variables)
+		},
+	}
+
+	cmd.Flags().StringVarP(&name, "name", "a", "", "application name")
+	cmd.Flags().StringVarP(&filename, "filename", "f", "", "path to the manifest file")
+	cmd.Flags().StringArrayVarP(&variables, "var", "v", []string{}, "set a pipeline variable (can be set more than once)")
```

Comment:
Not sure if this is needed or would be read from the manifest

Reply:
It is a nice addition. Not needed for the first version, but if it is already implemented I would keep it